src/App.js

```diff
@@ -1,16 +1,43 @@
 import React from 'react';
+//import { scryRenderedComponentsWithType } from 'react-dom/test-utils';
 import './App.css';
+//import ChatEntry from './components/ChatEntry';
+// import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [likeData, setLikeData] = useState(chatMessages);
+
+  const likeChat = (id) => {
+    setLikeData((likeData) => {
+      return likeData.map((entry) => {
+        if (entry.id === id) {
+          return { ...entry, liked: !entry.liked };
+        } else {
+          return entry;
+        }
+      });
+    });
+  };
```

Comment:
This looks great! Well done copying over the data before manipulating it and updating it!